src/metatracer.rs

```diff
@@ -64,19 +64,20 @@ impl Location {
 
 /// A meta-tracer.
 pub struct MetaTracer {
-    hot_threshold: AtomicU32
+    hot_threshold: HotThreshold
 }
 
 impl MetaTracer {
+    /// Create a new `MetaTracer` with default settings.
     pub fn new() -> Self {
-        Self {
-            hot_threshold: AtomicU32::new(DEFAULT_HOT_THRESHOLD)
-        }
+        Self::new_with_hot_threshold(DEFAULT_HOT_THRESHOLD)
     }
 
-    /// Update the hot threshold for this meta-tracer instance to `threshold`.
-    pub fn set_hot_threshold(&mut self, threshold: HotThreshold) {
-        self.hot_threshold.store(threshold, Ordering::Relaxed);
+    /// Create a new `MetaTracer` with a specific hot threshold `hot_threshold`.
```

Comment:
I'd probably kill the second backticked `hot_threshold` since it's obvious, but we are well into the realms of nitpicking now :)